scripts/run_e2e_tests_test.py

```diff
@@ -816,3 +828,31 @@ def test_do_not_run_with_test_suite_not_present_in_suites_list(self):
 
         with self.assertRaisesRegex(SystemExit, '1'):
             run_e2e_tests.main(args=['--suite', 'never'])
+
+    def test_do_not_run_with_test_non_mobile_suite_in_mobile_mode(self):
+        self.exit_stack.enter_context(self.swap_with_checks(
+            run_e2e_tests, 'is_oppia_server_already_running', lambda *_: False))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            run_e2e_tests, 'install_third_party_libraries', lambda _: None,
+            expected_args=[(False,)]))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            run_e2e_tests, 'build_js_files', lambda *_, **__: None,
+            expected_args=[(True,)]))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_elasticsearch_dev_server', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_firebase_auth_emulator', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_dev_appserver', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_redis_server', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_portserver', mock_managed_process))
+        self.exit_stack.enter_context(self.swap_with_checks(
+            servers, 'managed_cloud_datastore_emulator', mock_managed_process))
+        self.exit_stack.enter_context(self.swap(
+            flake_checker, 'check_if_on_ci', lambda: True))
+
+        with self.assertRaisesRegex(
+            SystemExit, '1'):
+            run_e2e_tests.main(args=['--mobile', '--suite', 'collections'])
```

Comment:
```suggestion
        with self.assertRaisesRegex(SystemExit, '1'):
            run_e2e_tests.main(args=['--mobile', '--suite', 'collections'])
```

Having the end of the `with` statement and the following block at the same indentation level makes the code hard to read

Reply:
Done